framework/src/base/MooseApp.C

```diff
@@ -1709,6 +1709,25 @@ MooseApp::hasRestartableMetaData(const std::string & name,
   }
 }
 
+RestartableDataValue &
+MooseApp::getRestartableMetaData(const std::string & name,
+                                 const RestartableDataMapName & metaname,
+                                 THREAD_ID tid) const
+{
+  if (!metaname.empty() && tid != 0)
```

Comment:
why !metaname.empty() ?
If there s no name dont we just get a default map? 
not super familiar

Reply:
I took this directly from `registerRestartableData`. However, looking at the logic flow, the following lines should return an error if metaname corresponds to an empty map since it will try to look for `name` within the `metaname` map and hit the error message. 

```
  // Get metadata reference from RestartableDataMap and return a (non-const) reference to its value
  auto & restartable_data_map = getRestartableDataMap(metaname);
  auto iter = restartable_data_map.find(name);
  if (iter == restartable_data_map.end())
    mooseError("Unable to find RestartableDataValue object with name " + name +
               " in RestartableDataMap");
```

Removed that extra condition in the if statement since it not needed